app/page/info/tokens.tsx

```diff
@@ -0,0 +1,229 @@
+import { faCaretDown, faXmark } from "@fortawesome/free-solid-svg-icons"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import * as ethers from "ethers"
+import { useState, type ChangeEvent, type FormEvent } from "react"
+
+import { useProviderContext } from "~app/context/provider"
+import { getChainName } from "~app/page/info/index"
+import { useAccount, useAction, useTokens } from "~app/storage"
+import type { BigNumberish, HexString } from "~typing"
+
+export function Tokens() {
+  const tokens = useTokens()
+  const account = useAccount()
+  const explorerUrl = `https://${getChainName(account.chainId)}.etherscan.io/`
+
+  const [isTokenModalOpened, setIsTokenModalOpened] = useState<boolean>(false)
+  const toggleAccountModal = () => setIsTokenModalOpened(!isTokenModalOpened)
+
+  return (
+    <div>
+      <div className="flex-col justify-center items-center h-auto p-3 border-0 rounded-lg text-base">
+        Tokens:
+        {tokens &&
+          tokens.map((token, index) => {
+            return (
+              <div key={index}>
+                <a
+                  href={`${explorerUrl}token/${token.address}`}
+                  target="_blank">
+                  {`${token.symbol}`}
+                </a>
+                {` ${parseFloat(
+                  ethers.formatUnits(
+                    ethers.toBeHex(token.balance),
+                    ethers.toNumber(token.decimals)
+                  )
+                ).toFixed(6)}`}
+              </div>
+            )
+          })}
+      </div>
+      <div className="col-span-3 cursor-pointer" onClick={toggleAccountModal}>
+        <span>Import Tokens</span>
+        <FontAwesomeIcon icon={faCaretDown} className="ml-2" />
+      </div>
+      {isTokenModalOpened && <TokenModal onModalClosed={toggleAccountModal} />}
+    </div>
+  )
+}
+
+function TokenModal({ onModalClosed }: { onModalClosed: () => void }) {
+  const { provider } = useProviderContext()
+  const { importToken } = useAction()
+  const account = useAccount()
+
+  const [tokenAddress, setTokenAddress] = useState<HexString>("")
+  const [tokenName, setTokenName] = useState<string>("")
+  const [tokenSymbol, setTokenSymbol] = useState<string>("")
+  const [tokenDecimals, setTokenDecimals] = useState<string>("")
+  const [firstOpenTokenModal, setFirstOpenTokenModal] = useState<boolean>(true)
+  const [invalidTokenAddress, setInvalidTokenAddress] = useState<boolean>(true)
```

Comment:
Why not set `invalidTokenAddress` initial to false, and remove `firstOpenTokenModal`?

Reply:
For submit button, we can add additional check about all fields are not empty.

Reply:
> Why not set invalidTokenAddress initial to false, and remove firstOpenTokenModal?

If `invalidTokenAddress` is true and `firstOpenTokenModal` is removed, the token input will show an error border when the user first enters this page.
Let me think about how to improve this...

Reply:
I said default `invalidTokenAddress` to `false`